pkg/controller/v1alpha2/applicationconfiguration/render.go

```diff
@@ -426,3 +429,33 @@ func addDataInputsToDAG(dag *dependency.DAG, ins []v1alpha2.DataInput, obj *unst
 		dag.AddSink(in.ValueFrom.DataOutputName, obj, attaches, in.ToFieldPaths)
 	}
 }
+
+// GetTraitName return trait name
+func getTraitName(ac *v1alpha2.ApplicationConfiguration, componentName string,
+	ct *v1alpha2.ComponentTrait, t *unstructured.Unstructured) string {
+	var (
```

Comment:
I assume we can check `t.GetName()` first, if this is not null, just return it. That's also the traitName specified by user.

Reply:
done